### PR TITLE
 Add `deviceId` and `latest` query parameters 

### PR DESCRIPTION
**Pull request by pazaan**
`deviceId` query parameter allows filtering of a user's deviceIDs
`lastest` query parameter will return an array of only the most recent
of each `type` that matches the other query parameter filters.

Removes previous addition of `sort` and `filter` query parameters.

- Fixes https://trello.com/c/CYFYBsQS
- Fixes https://trello.com/c/5GE2xbcZ

### CODE REVIEW

**Comment by pazaan** [comment]
[Add/update platform data API to return latest records by data type, sub type, and delivery type](https://trello.com/c/CYFYBsQS/642-add-update-platform-data-api-to-return-latest-records-by-data-type-sub-type-and-delivery-type)

**Comment by pazaan** [comment]
[[Backend] Backend changes needed for Trello #546](https://trello.com/c/5GE2xbcZ/625-backend-backend-changes-needed-for-trello-546)

File: store/store.go [hunk 1]
```diff
@@ -340,10 +332,10 @@ func (d MongoStoreClient) HasMedtronicLoopDataAfter(userID string, date string)
 	defer session.Close()
 
 	query := bson.M{
-		"_active":        true,
-		"_userId":        userID,
-		"_schemaVersion": bson.M{"$gt": 0},
-		"time":           bson.M{"$gte": date},
+		"_active":                            true,
+		"_userId":                            userID,
+		"_schemaVersion":                     bson.M{"$gt": 0},
```
Comment (derrickburns) [inline store/store.go]: Should this use the input parameters limiting the schema version?

File: store/store.go [hunk 2]
```diff
@@ -400,12 +392,61 @@ func (d MongoStoreClient) GetDeviceData(p *Params) StorageIterator {
...
+					},
+				},
+			},
+			{
+				"$lookup": bson.M{
+					"from":         "deviceData",
+					"localField":   "groupId",
+					"foreignField": "_id",
+					"as":           "latest_doc",
+				},
```
Comment (derrickburns) [inline store/store.go]: Left join.  Nice.  Is there an index on the join field?

Comment (pazaan) [reply store/store.go]: There sure is! `_id` is the default primary key in MongoDB

File: store/store.go (hunk 1 above)
Comment (pazaan) [reply store/store.go]: `_schemaVersion` has always been internal. I don't even see why it's particularly useful for us. Maybe we should check what it's function is and remove it altogether?
Either way, I'd make a separate card for that. This PR doesn't change the behaviour at all.

Comment (darinkrauss) [reply store/store.go]: Just saw this in my GitHub feed. We can remove `_schemaVersion` as it really isn't used anymore. *However*, we need to delete all `_schemaVersion=0` from the database before we remove it. The `_schemaVersion=0` is from data prior to a "reset" and is, essentially, invalid.

**Pull request merged**